checker.cpp

```diff
@@ -1,17 +1,73 @@
-#include <assert.h>
-
-bool vitalsAreOk(float bpm, float spo2, float respRate) {
-  if(bpm < 70 || bpm > 150) {
-    return false;
-  } else if(spo2 < 90) {
-    return false;
-  } else if(respRate < 30 || respRate > 95) {
-    return false;
-  }
-  return true;
-}
-
-int main() {
-  assert(vitalsAreOk(80, 95, 60) == true);
-  assert(vitalsAreOk(60, 90, 40) == false);
-}
+#include <iostream>
+#include <cassert>
+
+using namespace std;
+
+class Alert
+{
+    public:
+    virtual void raiseAlert(const char* vitalName, const char* level) = 0;
+};
+
+class AlertWithSms: public Alert
+{
+    public:
+    virtual void raiseAlert(const char* vitalName, const char* level){
+        cout << "SMS: The " << vitalName << " is very " << level << endl;
+    }
+};
+
+class AlertWithSound: public Alert
+{
+    public:
+    virtual void raiseAlert(const char* vitalName, const char* level){
+        cout << "Sound: The " << vitalName << " is very " << level << endl;
+    }
+};
+
+class Vital{
+    private:
+    int lowLimit, highLimit;
+    char* vitalName;
+    Alert* alertSystem;
+    public:
+    Vital(){}
+    Vital(int x, int y, char* z, Alert* alerter): lowLimit(x), highLimit(y), vitalName(z), alertSystem(alerter) {}
+    void isInRange(int val){
+        if(val < lowLimit){
+            alertSystem->raiseAlert(vitalName, "Low");
+        }
+        if(val> highLimit){
+            alertSystem->raiseAlert(vitalName, "High");
+        }
+    }
+};
+
+
+
+class VitalsChecker{
+private:
+    Alert* alertSystem;
+    Vital vitals[3];
+public:
+    VitalsChecker(Alert* alerter):alertSystem(alerter)
+    {
+        vitals[0] = Vital(70, 150, "Pulse checker", alertSystem);
+        vitals[1] = Vital(90, 100, "spo2", alertSystem);
+        vitals[2] = Vital(30, 95, "respRate", alertSystem);
+    }
+    void checkVitals(float* vals){
```

Comment:
It is always better to pass the number of elements when using raw arrays.
Alternative is to use vector<float>. Do not assume the size of the passed array inside the function. It is easy for someone to add something in the caller and miss updating the function implementation.

Reply:
Clean solution 👍